tests/test_metrics.py

```diff
@@ -51,6 +51,19 @@ def test_validate_label_with_incorrect_values(self, label):
         with pytest.raises(ValueError):
             collector._validate_labels([label])
 
+    def test_collect_without_labels(self):
+        counter = create_counter('name')
+        samples = counter._collector.collect()
+        assert len(samples) == 1
+
+    def test_collect_with_labels(self):
```

Comment:
Thank you for tests, beyond test coverage it help better understand the code 🙏

Reply:
Indeed, tests were very helpful!